run/types/DeviceWrapper.ts

```diff
@@ -298,17 +306,144 @@ export class DeviceWrapper {
+      includeScore: true,
+    });
+
+    const results = fuse.search(stripPrefix(selector));
+
+    if (results.length > 0 && results[0].score !== undefined && results[0].score < threshold) {
+      const match = results[0].item;
+      const confidence = ((1 - results[0].score) * 100).toFixed(2);
+
+      // Sometimes the element is just not on screen yet - proceed.
+      if (match.strategy === strategy && match.originalSelector === selector) {
+        return null;
+      }
+      this.log(
+        `Original locator ${strategy} "${selector}" not found. Test healed with ${match.strategy} "${match.selector}" (${confidence}% match)`
+      );
+      this.testInfo.annotations.push({
+        type: 'healed',
+        description: ` ${strategy} "${selector}" ➡ ${match.strategy} "${match.selector}" (${confidence}% match)`,
+      });
+      return {
+        strategy: match.strategy,
+        selector: match.originalSelector,
+      };
+    }
+
+    return null;
+  }
+
   public async findElement(strategy: Strategy, selector: string): Promise<AppiumNextElementType> {
-    return this.toShared().findElement(strategy, selector) as Promise<AppiumNextElementType>;
+    try {
+      return await (this.toShared().findElement(
+        strategy,
+        selector
+      ) as Promise<AppiumNextElementType>);
+    } catch (originalError) {
+      // Only try healing for id/accessibility id selectors
+      // In the future we can think about extracting values from XPATH etc.
+      if (strategy !== 'accessibility id' && strategy !== 'id') {
+        throw originalError;
+      }
+
+      const best = await this.findBestMatch(strategy, selector);
+
+      if (best) {
+        return await (this.toShared().findElement(
+          best.strategy,
+          best.selector
+        ) as Promise<AppiumNextElementType>);
+      }
+
+      throw originalError;
+    }
   }
 
   public async findElements(
     strategy: Strategy,
     selector: string
   ): Promise<Array<AppiumNextElementType>> {
-    return this.toShared().findElements(strategy, selector) as Promise<
-      Array<AppiumNextElementType>
-    >;
+    try {
+      const elements = await (this.toShared().findElements(strategy, selector) as Promise<
+        Array<AppiumNextElementType>
+      >);
+      if (elements && elements.length > 0) {
+        return elements;
+      }
+      throw new Error('No elements found');
+    } catch {
+      // Only try healing for id/accessibility id selectors
+      // In the future we can think about extracting values from XPATH etc.
+      if (strategy !== 'accessibility id' && strategy !== 'id') {
+        return [];
+      }
+
+      const best = await this.findBestMatch(strategy, selector);
+
+      if (best) {
+        try {
+          return await (this.toShared().findElements(best.strategy, best.selector) as Promise<
+            Array<AppiumNextElementType>
+          >);
+        } catch {
+          return [];
+        }
+      }
+
+      return [];
```

Comment:
same comment.

I guess that might not be needed if we always check that findElements returns something, wherever we call it?